src/server/ua_services_discovery.c

```diff
@@ -379,8 +379,13 @@ process_RegisterServer(UA_Server *server, UA_Session *session,
             return;
         }
         memcpy(filePath, requestServer->semaphoreFilePath.data, requestServer->semaphoreFilePath.length );
+#ifdef UNDER_CE
```

Comment:
In the PR #2042 I also added a new define for `UA_fileExists`.

You can redefine that one in your `ua_architecture.h` file with something like:

```
#define UA_fileExists(X) ({ \
                FILE *fp = fopen(X,"rb"); \
                bool exists = (fp==NULL); \
                if(fp) \
                    fclose(fp); \
                exists; \
})
```

Reply:
@Pro unfortunately this won't compile with my VS2017. It always says "undeclared identifier"

Reply:
I have also tested this but this wont work either. :-/ The statement expression seems to be a GCC feature which isn't supported in Visual Studio compiler.

Reply:
then as an alternative you can use your own function to your main architecture header:

```
UA_Boolean UA_fileExists(const char* path) {
                FILE *fp = fopen(X,"rb");
                UA_Boolean exists = (fp==NULL);
                if(fp)
                    fclose(fp);
                return exists;
}
```
That should work too and it will automatically disable the other define since that one is checking if `UA_fileExists` is already defined or not

Reply:
okay i will try this :)